pkg/controller/kibana/stackmon/sidecar.go

```diff
@@ -69,8 +69,8 @@ func Filebeat(client k8s.Client, kb kbv1.Kibana) (stackmon.BeatSidecar, error) {
 // WithMonitoring updates the Kibana Pod template builder to deploy Metricbeat and Filebeat in sidecar containers
 // in the Kibana pod and injects the volumes for the beat configurations and the ES CA certificates.
 func WithMonitoring(client k8s.Client, builder *defaults.PodTemplateBuilder, kb kbv1.Kibana) (*defaults.PodTemplateBuilder, error) {
-	// no monitoring defined, skip
-	if !monitoring.IsDefined(&kb) {
+	// no monitoring defined or yet configured, skip
+	if !monitoring.IsDefined(&kb) || !monitoring.AreAssocConfigured(&kb) {
```

Comment:
Given that we use this combined condition multiple times throughout the code base I wonder if it would make sense to have a single method 

```golang
func IsReconcilable(resource HasMonitoring) bool {
    return IsDefined(resource) && AreAssocConfigured(resource)
}
```

and then 
```suggestion
	if !monitoring.IsReconcilable(&kb) {
```

Reply:
Good idea.